Fix parallel tests by only testing the current optimization level

ppisar opened the pull request:
https://github.com/zeevt/csnappy/issues/39

zeevt commented:
I'll accept patches to fix the ability to run tests in parallel, but the tests with extra CFLAGS are needed. I want to know that regardless of which common CFLAGS override a user uses, the resulting binary is correct. So that if I use -O2 but a bug is only exposed by -O3, the tests will catch it. It's ok if the normal "make test" only uses the user's current CFLAGS, and "make test_extra_cflags" which is not part of "make test" tries the different CFLAGS.

ppisar committed:
Fix parallel tests by only testing the current optimization level

Testing other optimization levels is available under a new
test_extra_cflags Make target.

https://github.com/zeevt/csnappy/issues/39

ppisar commented:
Ok. I added test_extra_cflags target for only performing the different CFLAGS. Or did you mean that test_extra_cflags should perform both different CFLAGS tests and user's CFLAGS tests?. I.e. test_extra_cflags to be a superset of test target?

I also corrected an invocation of make by $(MAKE) variable to pass make options and job server correctly.
